src/mgopurge/trim_test.go

```diff
@@ -223,3 +223,65 @@ func (s *TrimSuite) TestTrimMultiDocWithExtras(c *gc.C) {
 	c.Assert(err, jc.ErrorIsNil)
 	c.Check(count, gc.Equals, 3)
 }
+
+func (s *TrimSuite) TestTrimMultiDocWithStash(c *gc.C) {
+	stash := s.db.C("txns.stash")
+	err := s.runner.Run([]txn.Op{{
+		C:      s.coll.Name,
+		Id:     0,
+		Insert: bson.M{"foo": "bar"},
+	}}, "", nil)
+	c.Assert(err, jc.ErrorIsNil)
+	// queue up a bunch of txns
+	txn.SetChaos(txn.Chaos{
+		KillChance: 1,
+		Breakpoint: "set-applying",
+	})
+	defer txn.SetChaos(txn.Chaos{})
+	ops := []txn.Op{{
+		C:      s.coll.Name,
+		Id:     0,
+		Update: bson.M{"$set": bson.M{"foo": "baz"}},
+	}, {
+		C:      s.coll.Name,
+		Id:     1,
+		Insert: bson.M{"new": "stuff"},
+	}}
+	for i := 0; i < 50; i++ {
+		c.Assert(s.runner.Run(ops, "", nil), gc.Equals, txn.ErrChaos)
+	}
+	var result bson.M
+	err = s.coll.FindId(0).One(&result)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(result["foo"], gc.Equals, "bar")
+	c.Check(result["txn-queue"], gc.HasLen, 51)
+	err = stash.FindId(bson.D{{"c", s.coll.Name}, {"id", 1}}).One(&result)
+	// if err != nil {
```

Comment:
No. At one point I was using bson.M for the query, but it seems the query is order sensitive (c, id) != (id, c), and it would occasionally fail. I ended up fixing the query instead of needing to handle the error. Removed.